### PR TITLE
Detect presence of policy tags in destination table.

### PR DESCRIPTION
**Pull request by agrawal-siddharth**
Revise restriction on writeAtLeastOnce.
Add integration tests to verify preservation of policy tags.

### CODE REVIEW

File: README-template.md [hunk 1]
```diff
@@ -545,7 +545,7 @@ word-break:break-word
     guarantee than exactly once. This is suitable for streaming scenarios
     in which data is continuously being written in small batches.
        <br/>(Optional. Defaults to <code>false</code>)
-       <br/><i>Supported only by the `DIRECT` write method and mode is <b>NOT</b> `Overwrite`.</i>
+       <br/><i>Supported only by the `DIRECT` write method and mode is <b>NOT</b> `Overwrite` for tables containing policy tags.</i>
```
Comment (suryasoma) [inline README-template.md:548]: Can we add a new line for `policyTags` instead? this is confusing
Rather, change it to 

<br/><i>Supported only by the `DIRECT` write method.</i>
<br/><i><b>NOT</b> supported for tables containing policy tags, and the write mode is `Overwrite`.</i>

File: bigquery-connector-common/src/main/java/com/google/cloud/bigquery/connector/common/BigQueryUtil.java [hunk 2]
```diff
@@ -271,6 +272,28 @@ public static boolean schemaWritable(
         sourceSchema.getFields(), destinationSchema.getFields(), enableModeCheckForSchemaFields);
   }
 
+  /**
+   * Checks whether at least one policy tag is present.
+   *
+   * @param schema the schema to check
+   * @return true if at least one policy tag is present, false otherwise
+   */
+  public static boolean schemaHasPolicyTags(Schema schema) {
```
[the remark this answers is not in the document]
Comment (davidrabinowitz) [reply bigquery-connector-common/src/main/java/com/google/cloud/bigquery/connector/common/BigQueryUtil.java:281]: Schema is not nested, but Field is. No recursion is needed, just two collections - current fields and next fields

[the remark this answers is not in the document]
Comment (suryasoma) [reply bigquery-connector-common/src/main/java/com/google/cloud/bigquery/connector/common/BigQueryUtil.java:281]: my bad, that is what I meant(typo), we should check the same, if the field has subFields

**Pull request closed without merging**